### PR TITLE
Support ansible provision mode for remote playbook

### LINKED ISSUE #2282: Additional provisioning formats beyond shell scripts
### Description

Currently we are using shell scripts, for the standard provisioning

It could also be useful to support other means, such as Ansible?

```yaml
provision:
- mode: ansible
  playbook: playbook.yaml
```

It would create an inventory, and then call `ansible-playbook`

* Host ("control node") requirements: Ansible

* Guest ("managed node") requirements: Python

See also:

* https://docs.ansible.com/ansible/latest/scenario_guides/guide_vagrant.html

* https://developer.hashicorp.com/vagrant/docs/provisioning/ansible ([v2.3.7](https://www.hashicorp.com/license-faq#products-covered-by-bsl))

```
╭─────────╮
⟨ ansible ⟩
╰─────────╯
        ╲   ^__^
         ╲  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
```

### PR DESCRIPTION
**Pull request by afbjorklund**
Closes #2282

~~Needs docs.~~

----

lima.yaml (based on debian.yaml)
```yaml
...

provision:
- mode: ansible
  playbook: playbook.yaml
```

inventory.yaml (auto-generated)
```yaml
lima:
  hosts:
    ansible:
      ansible_connection: ssh
      ansible_host: lima-ansible
      ansible_ssh_common_args: -F /home/anders/.lima/ansible/ssh.config
```

playbook.yaml
```yaml
- name: My first play
  hosts: all
  tasks:
   - name: Ping my hosts
     ansible.builtin.ping:

   - name: Print message
     ansible.builtin.debug:
      msg: Hello world
```

https://docs.ansible.com/ansible/latest/getting_started/get_started_playbook.html

```
INFO[0021] [hostagent] The final requirement 1 of 1 is satisfied 
INFO[0022] Waiting for ansible playbook "playbook.yaml" 
 ______________________
< PLAY [My first play] >
 ----------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

 ________________________
< TASK [Gathering Facts] >
 ------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

[WARNING]: Platform linux on host ansible is using the discovered Python
interpreter at /usr/bin/python3, but future installation of another Python
interpreter could change this. See https://docs.ansible.com/ansible/2.9/referen
ce_appendices/interpreter_discovery.html for more information.
ok: [ansible]
 ______________________
< TASK [Ping my hosts] >
 ----------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [ansible]
 ______________________
< TASK [Print message] >
 ----------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ok: [ansible] => {
    "changed": false,
    "msg": "Hello world"
}
 ____________
< PLAY RECAP >
 ------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

ansible                    : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   

INFO[0024] READY. Run `limactl shell ansible` to open the shell. 
```

### CODE REVIEW

File: pkg/store/filenames/filenames.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ const (
 	VzIdentifier       = "vz-identifier"
 	VzEfi              = "vz-efi"           // efi variable store
 	QemuEfiCodeFD      = "qemu-efi-code.fd" // efi code; not always created
+	InventoryYAML      = "inventory.yaml"   // ansible inventory
```
Comment (AkihiroSuda) [inline pkg/store/filenames/filenames.go:59]: Could this be prefixed with `ansible-` ?

File: pkg/start/start.go [hunk 2]
```diff
@@ -291,6 +291,10 @@ func watchHostAgentEvents(ctx context.Context, inst *store.Instance, haStdoutPat
 				return true
 			}
 
+			if xerr := runAnsibleProvision(ctx, inst); xerr != nil {
```
Comment (AkihiroSuda) [inline pkg/start/start.go:294]: Can we have an integration test in https://github.com/lima-vm/lima/blob/master/hack/test-templates/test-misc.yaml ?

**Review by balajiv113 — APPROVED** [review]
LGTM 👍

**Comment by RafalSkolasinski** [comment]
Is this feature officially supported? Coverage in docs seems to be minimal and couldn't find anything in the release notes. Cheers.

**Pull request merged**